### PR TITLE
Upgrade to yarn v4

### PR DESCRIPTION
**Pull request by smallsaucepan**
Turf was using a very old version of yarn (v1), with v4 being the most recently released. Upgrading to a newer version (apart from v1 eventually not being supported) we get access to some monorepo sugar - "workspace:^" instead of "v7.0.0-alpha.2" etc, and first class yarn workspaces commands. There might be some npm deployment related changes, though this has been around long enough they should be well documented. Tagging @mfedderly for a review regardless.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Have read [How To Contribute](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#how-to-contribute).
- [x] Run `npm test` at the sub modules where changes have occurred.
- [x] Run `npm run lint` to ensure code style at the turf module level.

### CODE REVIEW

**Comment by smallsaucepan** [comment]
Drat. Looks like the node-setup github action can't enable corepack, which we need to install the newer yarn version in the CI environment. Going to move this PR in to draft status until that is possible or I can find another way to press on 😩

Dependent on PR https://github.com/actions/setup-node/pull/901

**Comment by mfedderly** [comment]
For what its worth, I don't think we are tied to yarn for any particular reason (I moved us there because we didn't even have a node package lock at the time). I've mostly moved over to pnpm if that helps make this PR easier.

File: packages/turf-boolean-within/index.ts [hunk 1]
```diff
@@ -138,20 +138,18 @@ function isMultiPointOnLine(multiPoint: MultiPoint, lineString: LineString) {
...
     }
-    if (!oneInside) {
-      isInside = booleanPointInPolygon(multiPoint.coordinates[i], polygon, {
-        ignoreBoundary: true,
-      });
-    }
+
+    isInside = booleanPointInPolygon(multiPoint.coordinates[i], polygon, {
+      ignoreBoundary: true,
+    });
```
Comment (mfedderly) [inline packages/turf-boolean-within/index.ts:152]: This change is a no-op (I guess?) but its confusing as part of the yarn upgrade. It also might be a step away from a better version where we can skip the second booleanPointInPolygon? Not really sure what the original intent was of the oneInside boolean.

**Review by mfedderly — COMMENTED** [review]
Overall lgtm, happy to re-review once it builds.

File: packages/turf-boolean-within/index.ts (hunk 1 above)
Comment (smallsaucepan) [reply packages/turf-boolean-within/index.ts:152]: Me either. Yeah, it basically evaluated to ```if (true) ...```

Was an issue that didn't seem to justify its own PR, though happy to separate it out.

**Comment by smallsaucepan** [comment]
> For what its worth, I don't think we are tied to yarn for any particular reason (I moved us there because we didn't even have a node package lock at the time). I've mostly moved over to pnpm if that helps make this PR easier.

Well, I might see how this goes with pnpm instead and if that's smoother scrap this PR for one based around that tool. Thanks for clarifying.

**Comment by smallsaucepan** [comment]
Retiring in favour of PR #2555

**Pull request closed without merging**